source/Octopus.Client/IOctopusAsyncClient.cs

```diff
@@ -396,5 +354,7 @@ public interface IOctopusAsyncClient : IDisposable
         /// </summary>
         /// <returns></returns>
         Task SignOut();
+
+        bool IsAuthenticated { get; }
```

Comment:
Why do we need this new public method where we didn't need it before?

Reply:
Previously the client only gets the RootDocument (which allows anonymous) from the server during initialization. We are now also getting user and the spaces he/she can access to which requires authentication,  https://github.com/OctopusDeploy/OctopusClients/pull/318/files#diff-1b64c49f341db495d9a8b5a38fc1903cR281